src/get-or-create-render-context.js

```diff
@@ -7,53 +7,53 @@
 const vm = require("vm");
 // TODO(csilvers): try to get rid of the dependency on jsdom
 const jsdom = require("jsdom");
-const raf = require("raf");
 const cache = require("./cache.js");
 const profile = require("./profile.js");
 
-const runInContext = function(context, fn) {
-    return vm.runInContext("(" + fn.toString() + ")()", context);
-};
+class CustomResourceLoader extends jsdom.ResourceLoader {
+    fetch(url, options) {
+        // eslint-disable-next-line no-console
+        console.log("FETCHING: " + url);
+        return super.fetch(url, options);
+    }
+}
 
-const createRenderContext = function(jsPackages, pathToClientEntryPoint) {
-    const sandbox = {};
+const runInContext = function(jsdomContext, fnOrText) {
+    const script = typeof fnOrText === "function"
+        ? "(" + fnOrText.toString() + ")()"
+        : fnOrText;
+
+    if (typeof script !== "string") {
+        throw new Error("Must be a function or text");
+    }
+    return jsdomContext.runVMScript(new vm.Script(script));
+};
 
+const createRenderContext = function(locationUrl, jsPackages, pathToClientEntryPoint) {
     // A minimal document, for parts of our code that assume there's a DOM.
-    const doc = jsdom.jsdom(
+    const context = new jsdom.JSDOM(
         "<!DOCTYPE html><html><head></head><body></body></html>", {
-            // Forward console logs from inside the VM to the real console
-            virtualConsole: jsdom.createVirtualConsole().sendTo(console),
-            // CKEditor, and maybe other things, check for
-            // location.href, and expect it to be non-empty.
-            url:  "http://www.khanacademy.org",
+            url:  locationUrl,
+            runScripts: "dangerously",
+            resources: new CustomResourceLoader(),
+            pretendToBeVisual: true,
             features: {
                 FetchExternalResources: false,
                 ProcessExternalResources: false,
             },
         });
 
-    // Polyfill `requestAnimationFrame` to appease React.
-    raf.polyfill(doc.defaultView);
+    // This means we can run scripts inside the jsdom context.
+    context.run = fnOrText => runInContext(context, fnOrText);
 
-    // Copy everything from the jsdom object onto the sandbox.
-    // TODO(jlfwong): For some reason, copying doc.defaultView works, but
-    // using it directly as an arg to vm.createContext doesn't.
-    Object.keys(doc.defaultView).forEach(function(key) {
-        sandbox[key] = doc.defaultView[key];
-    });
-    sandbox.window = sandbox;
+    // Let's make sure our sandbox window is how we want.
+    const sandbox = context.window;
```

Comment:
Because JSDOM works very differently now. It sets up its own vm for example so the context is no longer accessible inside the VM as the VM's global object is window. Since window is the defaultView, it's already there.